File: ApplicationAlgorithm/test/cpp/unit11.cpp
#include <iostream>
using namespace std;

const int MAX = 1000;

int n, x[MAX];

int main() {
	cout << "N = "; cin >> n;
	x[0] = 1 , x[1] = 1;
	for (int i = 2; i <= n ; i++)
        x[i] = x[i - 1] + 2 * x[i - 2]; 
    cout << "So phuong an = " << x[n];
	return 0;
}

File: ApplicationAlgorithm/test/py/unit11.py
def function(n):
    a = [1, 1]
    for i in range(2, n + 1):
        a.append(a[i - 1] + 2 * a[i - 2])
    return a[n]

n = int(input("N = "))
print("So phuong an =", function(n))